operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,460 @@
+
+The following manifest represents a full example of a LokiStack with zone-aware data replication turned on using the `topology.kubernetes.io/zone` node label as a key to spread pods across zones and a replication factor of three
+
+```yaml
+apiVersion: loki.grafana.com/v1beta1
+kind: LokiStack
+metadata:
+  name: lokistack-dev
+spec:
+  size: 1x.small
+  storage:
+    secret:
+      name: test
+      type: s3
+  storageClassName: gp2
+  replication:
+    factor: 3
+    zones:
+    - topologyKey: topology.kubernetes.io/zone
+      maxSkew: 1
+```
+
+#### General constraints
+
+### Risks and Mitigations
+
+## Design Details
+
+### 1. Spreading the pods across failure domains using the Kubernetes scheduler (using the PodTopologySpreadConstraint)
+
+The loki components can be divided into the Write Path (Distributor, Ingester) and the Read Path (Query frontend, Querier, Ingester). In the first pass of the proposal do not distinguish between the two paths. See the `Drawbacks` section for more details on this.
+In this step, all componemts and their replicas are configured to spread over different zones. This can be done by making use of the `PodTopologySpreadConstraint` feature in Kubernetes. (See <https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/>)
+
+Commonly used node labels to identify domains are
+
+- topology.kubernetes.io/region
+- topology.kubernetes.io/zone
+- kubernetes.io/hostname
+
+Kubernetes makes a few assumptions about the structure of zones and regions:
+
+- regions and zones are hierarchical: zones are strict subsets of regions and no zone can be in 2 regions
+- zone names are unique across regions; for example region "africa-east-1" might be comprised of zones "africa-east-1a" and "africa-east-1b"
+
+The user needs to be aware of the node labels set to identify the different topology domains. Node read operations are generally made available to admin/developer users in OCP. This should be provided in the Lokistack CR topology key so that the `podTopologySpreadConstraint` can use this to schedule the pods accordingly.
+
+These are some of the checks needed before the deployment happens:
+
+- Check if the Topology key provided by the user matches a valid node topology domain label
+- Verify that the replication factor defined is less than the number of available zones
+
+If both these conditions are satisfied, deploy the lokistack components using the `PodTopologySpreadConstraint` definition in the deployments/statefulsets Pod template spec.
+
+### 2. Identifying the failure domain a pod is currently scheduled in (the two solutions discussed in length)
+
+The second step is to identify the domain the pod is scheduled on, to add the value into the loki configuration file(loki-config.yaml). There is no easy way to implement this since the Kubernetes Downward-API does not support exposing node labels within containers. (See issue <https://github.com/kubernetes/kubernetes/issues/40610>)
```

Comment:
```suggestion
The second step is to identify the domain the pod is scheduled in and then add this value into the Loki configuration file (`loki-config.yaml`). There is no easy way to implement this since the Kubernetes Downward-API does not support exposing node labels within containers. (see [Kubernetes issue](https://github.com/kubernetes/kubernetes/issues/40610))
```